File: src/main/java/me/pljr/pljrapispigot/config/TitleType.java
package me.pljr.pljrapispigot.config;

import me.pljr.pljrapispigot.builders.TitleBuilder;
import me.pljr.pljrapispigot.managers.ConfigManager;
import me.pljr.pljrapispigot.objects.PLJRTitle;
import org.bukkit.configuration.file.FileConfiguration;

import java.util.HashMap;

public enum TitleType {
    TELEPORT_TICKING(new TitleBuilder("§b§lTeleportation", "§7{time}§8s", 20, 40, 20).create()),
    TELEPORT_FAIL(new TitleBuilder("§b§lMove detected", "§cTeleportation canceled..", 20, 60, 20).create()),
    TELEPORT_TELEPORT(new TitleBuilder("§bTeleportation", "§7Teleporting..", 20, 60, 20).create());

    private static HashMap<TitleType, PLJRTitle> titleType;
    private final PLJRTitle defaultValue;

    TitleType(PLJRTitle defaultValue){
        this.defaultValue = defaultValue;
    }

    public static void load(ConfigManager config){
        FileConfiguration fileConfig = config.getConfig();
        titleType = new HashMap<>();
        for (TitleType title : values()){
            if (!fileConfig.isSet(title.toString())){
                config.setPLJRTitle(title.toString(), title.defaultValue);
            }else{
                TitleType.titleType.put(title, config.getPLJRTitle(title.toString()));
            }
        }
        config.save();
    }

    public PLJRTitle get(){
        return titleType.getOrDefault(this, defaultValue);
    }
}

File: src/main/kotlin/me/pljr/pljrapispigot/config/configuration/TitleType.kt
package me.pljr.pljrapispigot.config.configuration

import me.pljr.pljrapispigot.config.ConfigManager
import me.pljr.pljrapispigot.title.PLJRTitle
import me.pljr.pljrapispigot.title.TitleBuilder
import net.kyori.adventure.text.Component
import org.bukkit.configuration.file.FileConfiguration
import java.util.*

enum class TitleType(val defaultValue: PLJRTitle) {
    TELEPORT_TICKING(TitleBuilder(
        Component.text("&b&lTeleportation"),
        Component.text("&7{time}&8s"),
        20, 40, 20).create()),
    TELEPORT_FAIL(TitleBuilder(
        Component.text("&b&lTeleportation"),
        Component.text("&cTeleportation canceled.."),
        20, 60, 20).create()),
    TELEPORT_TELEPORT(TitleBuilder(
        Component.text("&b&lTeleportation"),
        Component.text("&7Teleporting.."),
        20, 60, 20).create());
    
    companion object {
        private val titleType = EnumMap<TitleType, PLJRTitle>(TitleType::class.java)

        fun load(config: ConfigManager) {
            val fileConfig: FileConfiguration = config.config
            titleType.clear()
            values().forEach {
                if (!fileConfig.isSet(it.toString())){
                    fileConfig[it.toString()] = it.defaultValue
                } else {
                    titleType[it] = config.getPLJRTitle(titleType.toString())
                }
            }
            config.save()
        }
    }

    fun get(): PLJRTitle {
        return titleType.getOrDefault(this, this.defaultValue)
    }
}